ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/State/Types.hs

```diff
@@ -37,6 +37,19 @@ import           Prelude
 -- | Generic hard fork state
 --
 -- This is used both for the consensus state and the ledger state.
+--
+-- By using a telescope, we will keep track of 'Past' information for eras
+-- before the current one:
+--
+-- > TZ currentByronState
+-- > TZ pastByronState $ TZ currentShelleyState
+-- > TZ pastByronState $ TS pastShelleyState $ TZ currentAllegraState
+-- > ...
+--
+-- Using 'Telescope.extend' and 'Telescope.retract' we are able to advance
+-- (_translate_) a @LedgerState@ from one era to the next/previous one. Using
+-- 'Telescope.align' or 'Telescope.alignExtend', we will also put in sync
+-- multiple 'Telescope's.
```

Comment:
I'd maybe copy the blockchain intuitions for `extend`, `retract` and `align` literally, and put them in sections. Something like:

```
-- === extend
--
-- ...
--
-- === retract
--
-- ...
--
-- === align
--
-- ...
```

Reply:
Done